test/validating_inputs_test.exs

```diff
@@ -4,62 +4,76 @@ defmodule ValidatingInputsTest do
-    assert ValidatingInputs.validate_type("Molving", :last_name) == "Molving"
+    func = &ValidatingInputs.validate_type/2
+    result = capture_invalid_message(func, "D\nM", "daisy\nmolving", [:first_name, :last_name])
+    assert result =~ "must be at least two characters long"
   end 
 
   test "a zip code must be completely numerical" do
-    assert capture_io("12345", fn ->
-       ValidatingInputs.validate_type("ABC4E", :zip_code) 
-    end) =~ "zip code must be numerical"
-    assert ValidatingInputs.validate_type("12345", :zip_code) == "12345"
+    func = &ValidatingInputs.validate_type/2
+    result = capture_invalid_message(func, "ABC4E", "12345", [:zip_code])
+    assert result =~ "zip code must be numerical"
   end
 
   test "an employee id must start with two letters, then a hyphen, then four numbers" do
-    assert capture_io("ZZ-7890", fn ->
-      ValidatingInputs.validate_type("AA1234", :employee_id)
-    end) =~ "employee id must have two letters followed by a hyphen, then four numbers"
-    assert ValidatingInputs.validate_type("AA-1234", :employee_id) == "AA-1234"
+    func = &ValidatingInputs.validate_type/2
+    result = capture_invalid_message(func, "AA1234", "ZZ-7890", [:employee_id])
+    assert result =~ "employee id must have two letters followed by a hyphen, then four numbers"
   end
 
-  test "invalid_empty message is printed when field is empty" do
-    assert capture_io("retry", fn -> ValidatingInputs.validate_not_empty("", :first_name) 
-    end) == "That is an invalid input, first_name cannot be empty. Please try again: "
+  test "retry_for_empty_field message is printed when field is empty" do
+    func = &ValidatingInputs.validate_not_empty/2
+    result = capture_invalid_message(func, "\n\n\n", "daisy\nmolving\n12345\nRR-7382", [:first_name, :last_name, :zip_code, :employee_id])
+    assert result =~ "cannot be empty. Please try again: "
   end 
 
-  test "invalid_empty message takes an input" do
-    input = "retry"
-    assert capture_io([input: input, capture_prompt: false], fn ->
-      IO.write(ValidatingInputs.retry_for_empty_field(:first_name))
-    end) == "retry"
+  test "retry_for_empty_field takes an input" do
+    func = &ValidatingInputs.validate_not_empty/2
+    result = capture_user_input(func, [:first_name, :last_name, :zip_code, :employee_id], "daisy\nmolving\n12345\nAA-12345")
+    assert result =~ "daisy\nmolving\n12345\nAA-12345"
   end 
+  
+  test "runs through every category to validate inputs" do
+    func = &ValidatingInputs.run_validation_checks/0
+    result = capture_final_output(func, "\ndaisy\nm\nmolving\nzip-error\n12345\nid-error\nGG-6735")
+    assert result =~ "no errors found"
+  end
 
-  test "prints no errors message" do
-    assert capture_io(fn ->
-      ValidatingInputs.print_no_found_error_message
-    end) =~ "no errors found"
+  defp capture_output(func, categories) do
+    capture_io(fn ->
+      Enum.each(categories, fn(category) ->
```

Comment:
I think you can shrink this to Enum.each(categories, func) but I am not 100% sure.